mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -193,12 +207,19 @@ class KMeans private (
 
     val initStartTime = System.nanoTime()
 
-    val centers = if (initializationMode == KMeans.RANDOM) {
-      initRandom(data)
-    } else {
-      initKMeansParallel(data)
+    val centers = initialModel match {
+      case Some(kMeansCenters) => {
+        Array.tabulate(runs)(r => kMeansCenters.clusterCenters
```

Comment:
I do not think setInitialModel should do anything other than set initialModel.

When training/fitting happens, I think we should then examine initialModel.  If initialModel is set, then we should act as though runs = 1 (but not actually change that value) and also print the warning message.

The documentation should be added within this PR.